ddtrace/tracer/textmap.go

```diff
@@ -306,15 +317,8 @@ func (p *chainedPropagator) Extract(carrier interface{}) (ddtrace.SpanContext, e
 // provided by the given *spanContext. If it matches, then the tracestate
 // will be re-composed based on the composition of the given *spanContext,
 // but will include the non-DD vendors in the W3C trace context's tracestate.
-func (p *propagatorW3c) propagateTracestate(ctx *spanContext, carrier interface{}) {
-	w3cCtx, _ := p.Extract(carrier)
-	if w3cCtx == nil {
-		return // It's not valid, so ignore it.
-	}
-	if ctx.TraceID() != w3cCtx.TraceID() {
-		return // The trace-ids must match.
-	}
```

Comment:
This check is not accounted for in the new code. Is it okay?

Reply:
We need this check for both `propagateTracestate` and `overrideDatadogParentID` so I moved it to `chainpropagator.Extract()`. This method is only called in one place so I think this is fine.

Reply:
I also made one small adjustment. Previously we were only validating the lower order bits in the trace_id. This was incorrect. We need to validate the full 128 bit trace id. This is a small minor fix. I can't see a scenario where traces have the same low order bits but different higher order bits (this would be extremely rare event or due to a bug in an upstream tracer).